relevanceai/operations_new/ops.py

```diff
@@ -333,3 +340,114 @@ def split_sentences(
             ),
         )
         return
+
+    @track
+    def cluster(
+        self,
+        model: Any = None,
+        alias: Optional[str] = None,
+        vector_fields: List[str] = None,
+        filters: Optional[list] = None,
+        include_cluster_report: bool = True,
+        **kwargs,
+    ):
+        """
+        Run clustering on your dataset.
+
+        Example
+        ----------
+
+        .. code-block::
+
+            from sklearn.cluster import KMeans
+            model = KMeans()
+
+            from relevanceai import Client
+            client = Client()
+            ds = client.Dataset("sample")
+            cluster_ops = ds.cluster(
+                model=model, vector_fields=["sample_vector_"],
+                alias="kmeans-8"
+            )
+
+        Parameters
+        ------------
+
+        model: Union[str, Any]
+            Any K-Means model
+        vector_fields: List[str]
+            A list of possible vector fields
+        alias: str
+            The alias to be used to store your model
+        cluster_config: dict
+            The cluster config to use
+            You can change the number of clusters for kmeans using:
+            `cluster_config={"n_clusters": 10}`. For a full list of
+            possible parameters for different models, simply check how
+            the cluster models are instantiated.
+        """
+        from relevanceai.operations_new.cluster.ops import ClusterOps
+
+        ops = ClusterOps(
+            model=model,
+            alias=alias,  # type: ignore
+            vector_fields=vector_fields,  # type: ignore
+            verbose=False,
+            credentials=self.credentials,
+            dataset_id=self.dataset_id,
+            **kwargs,
+        )
+        documents = self.get_all_documents(filters=filters, select_fields=vector_fields)
```

Comment:
Does this filter for vector fields exist?

Reply:
This isn't filtering for vector fields, it's just selecting the vector fields. 
Adding separate filters for the vector fields.

Reply:
This should also check for valid filters